### PR TITLE
GEA Cardiology first floor

### PR DESCRIPTION
**Pull request by Ana00000**
- Created CardiologyFirstFloorMapUserControl
- Created ViewModel for cardiologys first floor

### CODE REVIEW

**Comment by Ana00000** [comment]
[(2) Display Cardiologys first floor [3]](https://trello.com/c/LXj1xpRD/37-2-display-cardiologys-first-floor-3)

File: GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+                       VerticalAlignment="Center"/>
+
+                <!--Room1 button-->
+                <Button x:Name="Room1Button"
+                        Background="#FF237793" 
+                        BorderBrush="Black"
+                        CommandParameter="room1"
+                        Height="Auto"
+                        Margin="254,122,763,453"
+                        Style="{StaticResource ResourceKey=DefaultBuildingButtonStyle}">
```
Comment (lemara98) [inline GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml]: Background, Height, and BorderBrush is the same in every button. I suggest you to make a new style in Styles.xaml with BasedOn property with value DefaultBuildingButtonStyle, and move these repeating properties there. By doing this you will significantly shorten the code.

File: GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+                <Image Source="/GraphicEditor;component/Resources/Icons/chair.png"
+                       Margin="392,320,747,328"/>
+                <Image Source="/GraphicEditor;component/Resources/Icons/chair.png"
+                       Margin="482,320,657,328"
+                       RenderTransformOrigin="0.5,0.5">
+                    <Image.RenderTransform>
+                        <TransformGroup>
+                            <ScaleTransform ScaleY="1" 
+                                            ScaleX="-1"/>
+                            <SkewTransform AngleY="0"
```
Comment (lemara98) [inline GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml:171]: Don't use this if it doesn't change anything

File: GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+
+                <!--Doors-->
+                <Rectangle Fill="#FF61CCF0"
+                           HorizontalAlignment="Left"
+                           Height="109"
+                           Margin="254,320,0,0" 
+                           Stroke="Black" 
+                           StrokeThickness="3" 
+                           VerticalAlignment="Top"
+                           Width="10"/>
```
Comment (lemara98) [inline GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml]: Make a style

File: GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml (hunk 2 above)
Comment (Ana00000) [reply GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml:171]: This part changes the position of an icon, it flips her.

File: GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml [hunk 4]
```diff
@@ -124,14 +111,66 @@
...
+
+                <!--Doors-->
+                <Rectangle Width="110"
+                           Margin="202,366,0,0" 
+                           Style="{StaticResource DoorRectangle}">
+                    <Rectangle.RenderTransform>
+                        <TransformGroup>
+                            <ScaleTransform/>
+                            <SkewTransform/>
+                            <RotateTransform Angle="90"/>
```
Comment (lemara98) [inline GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml:134]: Can you make this without these <ScaleTransform/> and other tags. They seem unnecesary? If not, can you maybe move these things to style like in folowing pattern: Rename "DoorRectangle" to "DoorVerticalRectangle" and make a new style based on the previous like "DoorHorizontalRectangle" but only thing that is different is this rotation?

File: GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml [hunk 5]
```diff
@@ -124,14 +111,66 @@
...
+                            <TranslateTransform/>
+                        </TransformGroup>
+                    </Rectangle.RenderTransform>
+                </Rectangle>
```
Comment (lemara98) [inline GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml]: Can you make this without these and other tags. They seem unnecesary? If not, can you maybe move these things to style like in folowing pattern: Rename "DoorRectangle" to "DoorVerticalRectangle" and make a new style based on the previous like "DoorHorizontalRectangle" but only thing that is different is this rotation?

File: GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml [hunk 6]
```diff
@@ -124,14 +111,66 @@
...
+                           Style="{StaticResource DoorRectangle}"/>
+                <Rectangle Margin="844,485,0,0" 
+                           Style="{StaticResource DoorRectangle}"/>
+                <Image x:Name="image_Copy" 
+                    HorizontalAlignment="Left" 
+                    Height="44" 
+                    Margin="272,511,0,0" 
+                    VerticalAlignment="Top"
+                    Width="51"
+                    Source="/GraphicEditor;component/Resources/Icons/plant.png" RenderTransformOrigin="-1.059,3.273"/>
```
Comment (lemara98) [inline GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml]: Source property is not in alphabetic order :)

**Review by Ana00000 — COMMENTED** [review]
It's necessary for rotation, but i can make DoorVerticalRectangle.

File: GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml (hunk 5 above)
Comment (Ana00000) [reply GraphicEditor/View/UserControls/CardiologyFirstFloorMapUserControl.xaml]: It's necessary for rotation, but i can make DoorVerticalRectangle.

File: GraphicEditor/Resources/Styles/RectangleStyles.xaml [hunk 7]
```diff
@@ -1,15 +1,24 @@
...
+
+    <Style x:Key="DoorVerticalRectangle" TargetType="Rectangle">
+        <Setter Property="Fill" Value="#FF61CCF0" />
+        <Setter Property="Stroke" Value="Black" />
+        <Setter Property="StrokeThickness" Value="3" />
+        <Setter Property="Height" Value="50" />
+        <Setter Property="HorizontalAlignment" Value="Left" />
+        <Setter Property="VerticalAlignment" Value="Top" />
+        <Setter Property="Width" Value="10" />
     </Style>
```
Comment (lemara98) [inline GraphicEditor/Resources/Styles/RectangleStyles.xaml]: <Style x:Key="DoorVerticalRectangle" 
           BasedOn="DoorHorizontalRectangle" 
           TargetType="Rectangle">

You should do something like this :)

**Review by lemara98 — COMMENTED** [review]
Please correct the style as commented

**Pull request merged**